File: IronVault/UI/ironvault_desktop/lib/models/suplement.dart
import 'package:ironvault_desktop/models/dobavljac.dart';
import 'package:ironvault_desktop/models/kategorija.dart';
import 'package:json_annotation/json_annotation.dart';

part 'suplement.g.dart';

@JsonSerializable()
class Suplement {
  int? suplementId;
  String? naziv;
  double? gramaza;
  String? opis;
  String? slika;
  double? cijena;
  double? prosjecnaOcjena;
  int? dobavljacId;
  Dobavljac? dobavljac;
  int? kategorijaId;
  Kategorija? kategorija;

  Suplement({this.suplementId, this.naziv});

  /// Connect the generated [_$PersonFromJson] function to the `fromJson`
  /// factory.
  factory Suplement.fromJson(Map<String, dynamic> json) =>
      _$SuplementFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$SuplementToJson(this);
}

File: IronVault/IronVault.Model/Models/Suplement.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace IronVault.Model.Models
{
    public class Suplement
    {
        public int SuplementId { get; set; }
        public string Naziv { get; set; }
        public float Cijena { get; set; }
        public float Gramaza { get; set; }
        public string Opis { get; set; }
        public string? Slika { get; set; }
        //public byte[]? SlikaThumb { get; set; }
        public string? StateMachine { get; set; }
        public double? ProsjecnaOcjena { get; set; }
        public int DobavljacId { get; set; }
        public virtual Dobavljac Dobavljac { get; set; } = null!;
        public int KategorijaId { get; set; }
        public virtual Kategorija Kategorija { get; set; } = null!;

    }
}
